library/connect.c

```diff
@@ -105,6 +107,11 @@ void on_write_completed(struct ziti_conn *conn, struct ziti_write_req_s *req, in
     }
 
     free(req);
+
+    if (conn->write_reqs == 0 && conn->state == CloseWrite) {
+        ZITI_LOG(DEBUG, "sending FIN");
```

Comment:
small nit but maybe this log statement belongs in the function so that if it's called elsewhere - it'll log?